instrument-starters/opentracing-spring-cloud-reactor-starter/pom.xml

```diff
@@ -71,5 +71,11 @@
       <version>${version.io.opentracing.contrib-opentracing-spring-web}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>javax.annotation</groupId>
+      <artifactId>javax.annotation-api</artifactId>
+      <version>1.3.2</version>
+      <optional>true</optional>
```

Comment:
why it is added?

Reply:
I couldn't build without it, using JDK11. The compiler then gives the following error:
```
[ERROR] .../io/opentracing/contrib/spring/cloud/reactor/ReactorTracingAutoConfiguration.java:[27,24] package javax.annotation does not exist
[ERROR] .../io/opentracing/contrib/spring/cloud/reactor/ReactorTracingAutoConfiguration.java:[116,6] cannot find symbol
[ERROR]   symbol:   class PreDestroy
[ERROR]   location: class io.opentracing.contrib.spring.cloud.reactor.ReactorTracingAutoConfiguration.HookRegisteringPostProcessor
```
I put it to optional so that the dependency is not added to the transitive dependencies. (Perhaps provided is better in this case?)

If I switch to JDK 8 it does compile without the pom change. So, should I remove this `pom.xml` change and compile with JDK8 only?

Reply:
That's fine